crates/runtime/src/actor/module/mod.rs

```diff
@@ -166,41 +167,46 @@ async fn instantiate(
     config: &Config,
     handler: impl Into<builtin::Handler>,
 ) -> anyhow::Result<Instance> {
-    let mut wasi = WasiCtxBuilder::new();
-    let wasi = wasi
-        .arg("main.wasm")
-        .context("failed to set argv[0]")?
-        .build();
-    let ctx = Ctx {
-        wasi,
-        wasmbus: wasmbus::Ctx::new(handler),
-    };
-
-    let mut store = wasmtime::Store::new(module.engine(), ctx);
-    let memory = wasmtime::Memory::new(
-        &mut store,
-        wasmtime::MemoryType::new(config.min_memory_pages, config.max_memory_pages),
-    )
-    .context("failed to initialize memory")?;
-    linker
-        .define_name(&store, "memory", memory)
-        .context("failed to define `memory`")?;
-    let instance = linker
-        .instantiate_async(&mut store, module)
-        .await
-        .context("failed to instantiate module")?;
-
-    let start = instance.get_typed_func(&mut store, "_start");
-    let guest_call = instance
-        .get_typed_func::<guest_call::Params, guest_call::Result>(&mut store, "__guest_call");
-    let (start, guest_call) = match (start, guest_call) {
-        (Ok(start), Ok(guest_call)) => (Some(start), Some(guest_call)),
-        (Ok(start), Err(_)) => (Some(start), None),
-        (Err(_), Ok(guest_call)) => (None, Some(guest_call)),
-        (Err(_), Err(e)) => {
-            bail!("failed to instantiate either  `_start`, or `__guest_call`: {e}")
-        }
-    };
+    let (store, guest_call, start) = spawn({
```

Comment:
Did you have to spawn this to get it to spread to all cores or did it work without it. The `for_each_concurrent` one seems like it could be a problem, but not sure what is happening here that a spawn would help fix

Reply:
I've rechecked from scratch to be sure, deleting /tmp/was* and redeploying with only one part of the change and not the other. It really needs the two parts to spread to 100% on all core. 
Could it be that the wasmtime module reference being the same across all instances, the store stick to the same task ?